salt/proxy/restconf.py

```diff
@@ -0,0 +1,222 @@
+"""
+Proxy Minion to manage Restconf Devices
+
+:codeauthor: Jamie (Bear) Murphy <jamiemurphyit@gmail.com>
+:maturity:   new
+:platform:   any
+
+Usage
+=====
+
+.. note::
+
+    To be able to use this module you need to enable to RESTCONF on your device
+    and having https enabled.
+
+    Cisco Configuration example:
+
+    .. code-block:: bash
+
+        switch# conf t
+        switch(config)# restconf
+        switch(config)# ip http secure-server
+
+.. note::
+
+    RESTCONF requires modern OS distributions.
+    This plugin has been written specifically to use JSON Restconf endpoints
+
+Pillar
+------
+
+The ``restconf`` proxy configuration requires the following parameters in order
+to connect to the network switch:
+
+transport: ``https``
+    Specifies the type of connection transport to use. Valid values for the
+    connection are ``http``, and  ``https``.
```

Comment:
This contradicts the previous note about having https enabled.

Reply:
It's TECHNICALLY possible to use http but it is against the restconf standard. However its not like network device makers tend to follow standards. For example using VIRL images of cisco devices you have no choice as they have no crypto support built in